pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -395,3 +429,52 @@ func (c *sshClient) runSSH(command string) Output {
 		Err:    err,
 	}
 }
+
+func removeUselessLinesFromCloudInitOutput(s string) string {
+	regexes := []string{
+		`^\s*\d+K [. ]+ \d+%.*\ds$`,                  //  10000K .......... .......... .......... .......... ..........  6%!M(MISSING) 1s
+		`^Get:\d+ https?://.* [kM]B.*`,               // Get:17 http://archive.ubuntu.com/ubuntu focal/universe Translation-en [5,124 kB[]`
+		`^Preparing to unpack \.\.\..*`,              // Preparing to unpack .../04-libx11-6_2%!a(MISSING)1.6.9-2ubuntu1.6_amd64.deb ...\r
+		`^Selecting previously unselected package.*`, // Selecting previously unselected package kubeadm.\r
+		`^Setting up .* \.\.\..*`,                    // Setting up hicolor-icon-theme (0.17-2) ...\r
+		`^Unpacking .* \.\.\..*`,                     // Unpacking libatk1.0-0:amd64 (2.35.1-1ubuntu2) ...\r
+	}
+
+	// Compile the regexes
+	compiledRegexes := make([]*regexp.Regexp, 0, len(regexes))
+	for _, re := range regexes {
+		compiled, err := regexp.Compile(re)
+		if err != nil {
+			msg := fmt.Sprintf("removeUselessLinesFromCloudInitOutput: Failed to compile regex %s: %v\n", re, err)
+			fmt.Println(msg)
+			return msg
+		}
+		compiledRegexes = append(compiledRegexes, compiled)
+	}
+
+	var output []string
+
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Check if the line matches any of the regexes
+		matches := false
+		for _, re := range compiledRegexes {
+			if re.MatchString(line) {
+				matches = true
+				break
+			}
+		}
+
+		if matches {
+			continue
+		}
+		output = append(output, line)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return fmt.Sprintf("Error reading string: %v\n", err)
```

Comment:
I think you didn't push the code yet we looked at together...

Reply:
yes, I pushed these changes now.